### ISSUE TITLE
Resolve closure compiler "bad type annotation" warnings in graph plugin

### ISSUE THREAD

**Issue by wchargin** [tensorflow/tensorboard#355]
The graph plugin has a number of type annotations of the form `@type {tf.graph.Node}`. These yield warnings in the closure compiler: "WARNING - Bad type annotation. Unknown type tf.graph.Node."

At first glance, the type does appear to exist and be declared correctly. Perhaps something gets lost in the vulcanization process?

We used to have 7 warnings of this form; due to 100a153d, we now have 9. It would be very nice if we could achieve warning-level-zero, and require that new code maintain this.

**Comment by chihuahua** [comment]
These are the warnings. It seems like any warning within an HTML file triggers a warning.

```
bazel run tensorboard:tensorboard -- --logdir=~/Desktop/mnist_with_functions --port=5060
INFO: Found 1 target...
INFO: From Vulcanizing /tensorboard.html:
2017-08-20 23:46:29.352 WARNING /tf-graph-info/tf-graph-icon.html.js:11: WARNING - Bad type annotation. Unknown type tf.graph.Node
           * @type {tf.graph.Node}
                    ^

2017-08-20 23:46:29.793 WARNING /tf-graph-info/tf-graph-icon.html.js:22: WARNING - Bad type annotation. Unknown type tf.graph.render.RenderNodeInfo
           * @type {tf.graph.render.RenderNodeInfo}
                    ^

2017-08-20 23:46:29.793 WARNING /tf-graph-info/tf-node-list-item.html.js:9: WARNING - Bad type annotation. Unknown type tf.graph.Node
           * @type {tf.graph.Node}
                    ^

2017-08-20 23:46:29.794 WARNING /tf-graph-info/tf-node-list-item.html.js:14: WARNING - Bad type annotation. Unknown type tf.graph.Node
           * @type {tf.graph.Node}
                    ^

2017-08-20 23:46:29.794 WARNING /tf-graph-op-compat-card/tf-graph-op-compat-list-item.html.js:9: WARNING - Bad type annotation. Unknown type tf.graph.Node
       * @type {tf.graph.Node}
                ^

2017-08-20 23:46:29.794 WARNING /tf-graph-op-compat-card/tf-graph-op-compat-list-item.html.js:14: WARNING - Bad type annotation. Unknown type tf.graph.Node
       * @type {tf.graph.Node}
                ^

2017-08-20 23:46:29.795 WARNING /tf-graph/tf-graph-scene.html.js:20: WARNING - Bad type annotation. Unknown type d3_zoom
    /** @type {d3_zoom} d3 zoom object */
               ^

2017-08-20 23:46:29.795 WARNING /tf-graph/tf-graph-scene.html.js:60: WARNING - Bad type annotation. Unknown type d3.scale.ordinal
     * @type {d3.scale.ordinal}
              ^

2017-08-20 23:46:29.796 WARNING /tf-graph/tf-graph-scene.html.js:67: WARNING - Bad type annotation. Unknown type tf.scene.Minimap
     * @type {tf.scene.Minimap}
              ^

2017-08-20 23:46:29.796 WARNING 0 error(s), 9 warning(s), 63.1% typed
Target //tensorboard:tensorboard up-to-date:
  bazel-bin/tensorboard/tensorboard
INFO: Elapsed time: 50.484s, Critical Path: 50.18s

INFO: Running command line: bazel-bin/tensorboard/tensorboard '--logdir=~/Desktop/mnist_with_functions' '--port=5060'

```

I would be in favor of removing the type annotations and explicitly commenting on the types.

**Comment by wchargin** [comment]
> I would be in favor of removing the type annotations and explicitly commenting on the types.

I support this as a last resort, but I suspect that it won't be necessary. Presumably, these types were once valid, but due to some change in the build process the closure compiler is no longer able to resolve them. The developers of this code thought that these type annotations were useful, so we should try to keep them.

(Possible exception of `d3_zoom`, which looks like it might not be a valid type.)

If in fact these types were actually just free-form text that never had any meaning—like `@type {a valid node}` or something—then removing them would certainly be the way to go. But the namespaced way in which they're written suggests to me that this is unlikely to be the case.

**Comment by chihuahua** [comment]
That is true. Based on internal code history, @dsmilkov introduced those annotations long ago.

One thing that might work is importing the types. Let me try that.

**Comment by dsmilkov** [comment]
+1 to removing `@type {...}` altogether. They are no longer valid annotations - they used to be very early on, when we used closure, before we migrated to typescript.

You can still mention the type but in free-form text.

**Comment by wchargin** [comment]
Okay—makes sense to me. Thanks!

**Comment by chihuahua** [comment]
Thank you for clarifying! In that case, I will proceed to remove annotations and explicitly mention type as did #393.

I tried including `<link rel="import" href="../tf-graph-common/tf-graph-common.html">` to no avail. I also tried importing importing individual classes like `import {Minimap} from "../tf-graph-common/minimap.ts.js";` but found it a bit odd how I was reaching into the innards of `tf-graph-common` while the point of that component was to gather a bunch of different types together for easy importing.
